Scripts/aws_stack_provision.ps1

```diff
@@ -1,103 +1,302 @@
 ## Powershell script for
 
 param (
-    [string] $lansa_version
+[parameter(Mandatory=$true)]    [string] $lansa_version,
+[parameter(Mandatory=$false)]   [string] $tag_lansa_version ## lansa version that would be used to tag cloned stack
+
 )
 
+Write-Host "Searching for VM with Lansa Version tag = $lansa_version"
 
 $EXISTING_INSTANCE_COUNT = (((Get-EC2Instance -Filter @{ Name="tag:LansaVersion"; Values=$lansa_version }).Instances).InstanceId).count
 
 if ($EXISTING_INSTANCE_COUNT -eq 1 ){
-  echo "VM with same lansa Version tag exist"
-  echo "Checking VM Status"
-  $INSTANCE_ID = ((Get-EC2Instance -Filter @{ Name="tag:LansaVersion"; Values=$lansa_version }).Instances).InstanceId
-  $INSTANCE_STATE = (((Get-EC2InstanceStatus -IncludeAllInstance $true -InstanceId $INSTANCE_ID).InstanceState).Name).Value
-
-  if ($INSTANCE_STATE -eq "running"){
-     echo "VM is already in running state"}
-  elseif ($INSTANCE_STATE -eq "stopped"){
-     echo "VM is in stopped state"
-     echo "Starting the VM"
-     $INSTANCE_ID = ((Get-EC2Instance -Filter @{ Name="tag:LansaVersion"; Values=$lansa_version }).Instances).InstanceId
-     Start-Sleep -Seconds 60
-     $NEW_STATUS = (((Start-EC2Instance -InstanceID $INSTANCE_ID).CurrentState).Name).Value
-     echo "Current status of VM is $NEW_STATUS"}
-  else{
-     echo "VM is neither Running nor stopped"
-     echo "VM status is $INSTANCE_STATE"}  }
+  
+	Write-Host "VM with Lansa Version tag = $lansa_version exist"
+	Write-Host "Checking VM Status"
+	$INSTANCE_ID = ((Get-EC2Instance -Filter @{ Name="tag:LansaVersion"; Values=$lansa_version }).Instances).InstanceId
+	$INSTANCE_STATE = (((Get-EC2InstanceStatus -IncludeAllInstance $true -InstanceId $INSTANCE_ID).InstanceState).Name).Value
+
+  	if ($INSTANCE_STATE -eq "running"){
+		Write-Host "VM is already in running state"}
+
+	elseif ($INSTANCE_STATE -eq "stopped"){
+     		Write-Host "VM is in stopped state"
+     		Write-Host "Starting the VM"
+     		$INSTANCE_ID = ((Get-EC2Instance -Filter @{ Name="tag:LansaVersion"; Values=$lansa_version }).Instances).InstanceId
```

Comment:
Why do you need to get the Instance ID again? They do not change between stopped and running.

Reply:
I have removed it now.